### PR TITLE
fs immer 2

### PR DESCRIPTION
**Pull request by songgao**
* [x] after release

### CODE REVIEW

File: shared/reducers/fs.tsx [hunk 1]
```diff
@@ -326,28 +353,35 @@ export default Container.makeReducer<FsGen.Actions, Types.State>(initialState, {
...
-      edit.set('status', Types.EditStatusType.Saving)
-    )
+    updateExistingEdit(draftState, action.payload.editID, draftEdit => {
+      draftEdit.status = Types.EditStatusType.Saving
+    })
   },
   [FsGen.discardEdit]: (draftState, action) => {
-    draftState.edits = draftState.edits.remove(action.payload.editID)
+    if (draftState.edits.has(action.payload.editID)) {
+      const edits = new Map([...draftState.edits])
```
Comment (chrisnojima) [inline shared/reducers/fs.tsx]: why not `const edits = new Map(draftState.edits)`? cause `new Map(undefined)`? if so maybe
`new Map(draftState.edits || [])`

Comment (chrisnojima) [reply shared/reducers/fs.tsx]: (and below)

File: shared/reducers/fs.tsx [hunk 2]
```diff
@@ -356,38 +390,51 @@ export default Container.makeReducer<FsGen.Actions, Types.State>(initialState, {
...
+    }
   },
   [FsGen.setDestinationPickerParentPath]: (draftState, action) => {
-    draftState.destinationPicker = draftState.destinationPicker.update('destinationParentPath', list =>
-      list.set(action.payload.index, action.payload.path)
-    )
+    if (draftState.destinationPicker.destinationParentPath[action.payload.index] !== action.payload.path) {
+      const destinationParentPath = [...draftState.destinationPicker.destinationParentPath]
+      destinationParentPath[action.payload.index] = action.payload.path
+      draftState.destinationPicker.destinationParentPath = destinationParentPath
```
Comment (chrisnojima) [inline shared/reducers/fs.tsx]: you don't need to copy to mutate arrays in immer
`draftState.destinationPicker.destinationParentPath[action.payload.index] = action.payload.path`

File: shared/reducers/fs.tsx (hunk 1 above)
Comment (songgao) [reply shared/reducers/fs.tsx]: Nope; I just didn't know! Will cahnge

**Pull request merged**